### PR TITLE
Console guidelines

### PR DESCRIPTION
**Pull request by mrhelmut**
I started drafting some guidelines to help prepare for porting games to consoles with MonoGame.

Review appreciated.

### CODE REVIEW

**Review by SimonDarksideJ — APPROVED** [review]

File: articles/preparing_for_consoles.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+- Add `<PublishAot>true</PublishAot>` to your `.csproj`.
+- Then run `dotnet publish` from the command-line/terminal.
+
+This will nativily compile your game in a fashion similar to consoles. The output executable will be in your output folder, within a sub-folder nammed `publish`.
+
+From there, you can try running this executable. If it does not start or crashes later on, you likely are using reflection or another unsupported feature in a AOT runtime.
+
+Native executables can debugged by:
+
+- Atarting an empty Visual Studio with no code.
```
Comment (dellis1972) [inline articles/preparing_for_consoles.md:36]: Typo, I assume it is supposed to be 'Starting'

File: articles/preparing_for_consoles.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+Many third party libraries heavily rely on using reflection or IL emit, this is a common practice for JSON or XML parsers for example.
+
+It is advised to choose very carefully the libraries that you are using when porting to consoles. If you do not select them with this in mind, you might run into a situation in which you will have to rewrite entire chunks of data handling.
+
+The best way to make sure if they will work, is to search if they are **"AOT-compatible"**, or try to compile with the `<EnableTrimAnalyzer>true</EnableTrimAnalyzer>` setting in your `.csproj` and check if there are any warnings related to those libraries.
+
+## Native libraries
+
+If you are using native libraries, make sure that they propose builds for consoles, or make sure that you can compile and run them on consoles yourself.
```
Comment (dellis1972) [inline articles/preparing_for_consoles.md:66]: `make sure that they propose builds for consoles` should be `make sure that they provide builds for consoles` ?

**Review by dellis1972 — CHANGES_REQUESTED** [review]
Looks good, some minor changes 👍

File: articles/preparing_for_consoles.md (hunk 1 above)
Comment (SimonDarksideJ) [reply articles/preparing_for_consoles.md:36]: Fixed

File: articles/preparing_for_consoles.md (hunk 2 above)
Comment (SimonDarksideJ) [reply articles/preparing_for_consoles.md:66]: Updated

**Review by dellis1972 — APPROVED** [review]

**Comment by mrhelmut** [comment]
I've fixed some typos.

If it's good to you, it's good to me. ✌

**Review by SimonDarksideJ — APPROVED** [review]

**Pull request merged**